### DISCUSSION TITLE
如何联动项目模块并debug项目

### DISCUSSION THREAD

**Discussion by wylovely** [rime/home#1520] [Q&A] [answered]
各位大佬，我想要深度学习一下这个项目，想启动并断点调试一下，梳理一下各模块的关系，但是我目前不知道如何将这几个项目联合起来debug，有没有大佬告诉我一下应该如何配置这些东西才能debug跑起来(本人没有cpp开发经验😂)

![image](https://github.com/rime/home/assets/50231506/3206a1cb-2b53-453d-8402-c088f0333457)

我把其他的一些打开会报不是可执行的dll错误

类似这种

![image](https://github.com/rime/home/assets/50231506/f32e3fa3-247a-4489-9e78-93fe149cd688)

我想要的效果是模块联动起来我打断点执行到哪个模块哪个模块就可以停住debug，不知道可不可以

目前尝试把TestWeaselIPC 跑起来了，加入WeaselUI，跑的时候说是RimeWithWeasel.lib没有，但我看到rime是单独的一个解决方案，不知道如何弄进来

在此先谢过各位大佬了！🙏🙏🙏

**Comment by ruoh6** [comment]
根据您发的报错，缺少rime.dll文件，需要编译librime。希望以上建议能够解决您的问题。:)

**Comment by fxliang** [comment, marked as answer]
项目本身不用修改就可以有办法构建debug版本的产出

`build.bat debug`即可构建除hant之外的产出用作调试用途，另外最新的代码中，tsf的hant配置已经不再构建，只有IME还有，影响不大。

另外，输入法调试的时候，要打开不同应用使用不同输入法，vs不用weasel, 不然调试有可能会都让输入全部卡死

此外，如果你完全没有C++经验，这个事情建议你不要做，太超过了
